background.js

```diff
@@ -0,0 +1,74 @@
+"use strict";
+
+// Add site to storage
+function addSite(tab, callback) {
+    checkSite(tab.url, function(allowed) {
+        if (!allowed) {
+            return;
+        }
+        // Get a favicon properly
+        if (!tab.favIconUrl || tab.favIconUrl.indexOf("chrome://theme") > -1) {
+            tab.favIconUrl = chrome.runtime.getURL("../img/favicon.png");
+        }
+        chrome.storage.local.get("sites", function(storage) {
+            var storage = storage["sites"] || [];
+            var site = { title: tab.title, faviconUrl: tab.favIconUrl, url: tab.url };
+            // Push site object into storage and save it
+            storage.push(site);
+            chrome.storage.local.set({sites: storage}, function() {
+                callback();
+            });
+        });
+    });
+}
+
+// Check site's url, whether note should be added or not
+function checkSite(url, callback) {
+    // URL may be undefined in some cases, GH #16
+    if (url === undefined) {
+        callback(false);
+        return;
+    }
+    chrome.storage.local.get("sites", function(storage) {
+        var sites = storage["sites"];
+        if (!sites) {
+            callback(false);
+            return;
+        }
+        var allowed = true;
+        for (let i=0; i<sites.length; i++) {
+            var siteUrl = sites[i].url;
+            if (siteUrl === url) {
+                allowed = false;
+                break;
+            }
+        }
+        callback(allowed);
+    });
+}
+
+// Remove site from storage
+function removeSite(url, callback) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: ['removeSite' is defined but never used.](https://www.codacy.com/app/tomastaro/notemark/file/1131180199/issues/source?bid=2884821&fileBranchId=2904179#l51)